arch/riscv/core/thread.c

```diff
@@ -259,6 +269,9 @@ FUNC_NORETURN void arch_user_mode_enter(k_thread_entry_t user_entry,
 	register void *a1 __asm__("a1") = p1;
 	register void *a2 __asm__("a2") = p2;
 	register void *a3 __asm__("a3") = p3;
+#if defined(CONFIG_THREAD_LOCAL_STORAGE
+	register void *tp __asm__("tp") = _current->tls;
```

Comment:
`register void *tp __asm__("tp") = (void *)_current->tls;`

Reply:
This hunk isn't necessary; tp gets set before we reach this point now.